### PR TITLE
Fix cross validation for KNN classifiers

### PR DESCRIPTION
**Pull request by AidenRushbrooke**
#### Reference Issues/PRs

Related to issue #596.

#### What does this implement/fix? Explain your changes.

Rolls back changes relating to the terms "distance_params" and "metric_params" to fix cross validation for KNN classifers.
Also includes setting the defaults for each parameter manually, if none have been provided.

#### Does your contribution introduce a new dependency? If yes, which one?

No
#### What should a reviewer concentrate their feedback on?

Are there better ways to fix this issue while keeping the same naming conventions
#### Any other comments?

Due to the way the classifier is cloned during cross validation, the distance metric was not being set correctly. This is because the distance functions still look for the parameters in metric_params, following the scikit-learn implementation. There may be better ways of resolving this issue whilst keeping the same naming convention, but I found this method the cleanest of options.

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/master/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/master/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Comment by TonyBagnall** [comment]
hi aiden, it fails one of my tests here, set "g":0.05 

 elif distance == wdtw_distance or distance == wddtw_distance:
            return {"g": 0}

**Comment by AidenRushbrooke** [comment]
Hi, just made that change,
Not sure about the other issue, I think its to do with metric_params being set when the test expects it to be empty when using default values.

**Comment by TonyBagnall** [comment]
> Hi, just made that change,
> Not sure about the other issue, I think its to do with metric_params being set when the test expects it to be empty when using default values.

could you send me the code you are using to cross validate on teams? ta

**Comment by mloning** [comment]
@AidenRushbrooke @TonyBagnall The tests fail because the hyper-parameter specified in the constructor is not stored in the object.
